libs/core/src/lib/popover/cdk-overlay/cdk-popover.component.ts

```diff
@@ -0,0 +1,395 @@
+ * The control is what will trigger the opening of the actual popover, which is called the body.
+ * By default, popovers are triggered by click. This can be customized through the *triggers* input.
+ * PopoverComponent is an abstraction of PopoverDirective.
+ */
+@Component({
+    selector: 'fd-cdk-popover, fd-popover',
+    templateUrl: './cdk-popover.component.html',
+    host: {
+        '[class.fd-popover-custom]': 'true',
+        '[attr.id]': 'id'
+    },
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    styleUrls: ['./cdk-popover.component.scss']
+})
+export class CdkPopoverComponent extends BasePopoverClass
+    implements AfterViewInit, OnInit, OnDestroy, OnChanges {
+
+    /** Reference to popover trigger element */
+    @Input()
+    trigger: ElementRef;
+
+    /** Whether the popover is disabled. */
+    @Input()
+    @HostBinding('class.fd-popover-custom--disabled')
+    disabled = false;
+
+    /** Id of the popover. If none is provided, one will be generated. */
+    @Input()
+    id: string = 'fd-popover-' + popoverUniqueId++;
+
+    /** @hidden */
+    @ViewChild('templateRef', { read: TemplateRef })
+    templateRef: TemplateRef<any>;
+
+    /** @hidden */
+    @ViewChild('container', { read: ViewContainerRef })
+    container: ViewContainerRef;
+
+    /** @hidden */
+    @ViewChild(CdkOverlayOrigin)
+    triggerOrigin: CdkOverlayOrigin;
+
+    /** position of arrow, passed to arrow element */
+    arrowPosition: string = null;
+
+    /** Additional style to put margin into body component, to give a place for arrow */
+    marginStyle: string = null;
+
+    /** @hidden */
+    private _initialised = false;
+
+    /** @hidden */
+    private _refresh$: Observable<void>;
+
+    /** @hidden */
+    private _eventRef: Function[] = [];
+
+    /** @hidden */
+    private _overlayRef: OverlayRef;
+
+    /** An RxJS Subject that will kill the data stream upon component’s destruction (for unsubscribing)  */
+    private readonly _onDestroy$: Subject<void> = new Subject<void>();
+
+    constructor(
+        private _renderer: Renderer2,
+        private _changeDetectorReference: ChangeDetectorRef,
+        private _overlay: Overlay,
+        private _viewportRuler: ViewportRuler,
+        @Optional() private _rtlService: RtlService
+    ) {
+        super();
+
+        /** Merge observables - close or destroy */
+        this._refresh$ = merge(this.isOpenChange, this._onDestroy$);
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        if (!this.scrollStrategy) {
+            this.scrollStrategy = this._overlay.scrollStrategies.reposition();
+        }
+
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this._initialised = true;
+        this._refreshTriggerListeners();
+        if (this.isOpen) {
+            this.open();
+        }
+    }
+
+    /** @hidden */
+    ngOnChanges(changes: SimpleChanges): void {
+        if (!this._initialised) {
```

Comment:
I think moving this to setters will save us from this additional "_initialised" flag.

Reply:
Nope, setters will trigger open/close method, before view/content have been initialised.